src/Workspaces/MSBuildTest/OpenProjectsTests.cs

```diff
@@ -0,0 +1,225 @@
+
+            // These templates matched your input: --language='vb', --type='project'
+            //
+            // Template Name                  Short Name           Language  Tags
+            // -----------------------------  -------------------  --------  ---------------
+            // Class Library                  classlib             C#,F#,VB  Common/Library
+            // Console App                    console              C#,F#,VB  Common/Console
+            // ...
+
+            var result = RunDotNet($"new list --type project --language {language}");
+            Assert.Equal(0, result.ExitCode);
+
+            var lines = result.Output
+                .Replace("\r\n", "\n")
+                .Split(new[] { "\n" }, StringSplitOptions.RemoveEmptyEntries);
+
+            TheoryData<string> templateNames = [];
+            var foundDivider = false;
+
+            foreach (var line in lines)
+            {
+                if (!foundDivider)
+                {
+                    if (line.StartsWith("----"))
+                    {
+                        foundDivider = true;
+                    }
+                    continue;
+                }
+
+                var columns = line.Split(new[] { "  " }, StringSplitOptions.RemoveEmptyEntries)
+                    .Select(c => c.Trim())
+                    .ToArray();
+                var templateShortName = columns[1].Split(',').First();
+
+                if (ExcludeMauiTemplates && templateShortName.StartsWith("maui"))
+                    continue;
+
+                templateNames.Add(templateShortName);
+            }
+
+            return templateNames;
+        }
+
+        private async Task AssertTemplateProjectLoadsCleanlyAsync(string templateName, string languageName, string[]? ignoredDiagnostics = null)
+        {
+            if (ignoredDiagnostics?.Length > 0)
+            {
+                TestOutputHelper.WriteLine($"Ignoring compiler diagnostics: \"{string.Join("\", \"", ignoredDiagnostics)}\"");
+            }
+
+            using var projectDirectory = (DisposableDirectory)SolutionDirectory;
+
+            var projectFilePath = GenerateProjectFromTemplate(projectDirectory.Path, templateName, languageName, TestOutputHelper);
+
+            await AssertProjectLoadsCleanlyAsync(projectFilePath, ignoredDiagnostics ?? []);
+
+            return;
+
+            static string GenerateProjectFromTemplate(string projectPath, string templateName, string languageName, ITestOutputHelper outputHelper)
+            {
+                var projectFilePath = GetProjectFilePath(projectPath, languageName);
+
+                CreateNewProject(templateName, projectPath, languageName, outputHelper);
+
+                return projectFilePath;
+            }
+
+            static string GetProjectFilePath(string projectPath, string languageName)
+            {
+                var projectName = Path.GetFileName(projectPath);
```

Comment:
This threw me off that we're using GetFileName as a way to actually ask for the directory name. Consider either adding a comment, or maybe just swap this out with:

```suggestion
                var projectName = new DirectoryInfo(projectPath).Name;
```

Reply:
That code is no longer what was intended. It worked better when I was making named subdirectories for each project. Will update it to just use templateName.

Reply:
haha, nope using `-o` definitely names the project after the output folder. Will take your suggestion after all.